### PR TITLE
8341553: Remove UseCompactObjectHeaders extra CDS archives

### PR DESCRIPTION
**Pull request by calvinccheung**
With this change, the compact object headers related CDS archives (`classes_coh.jsa` and `classes_nocoops_coh.jsa`) will not be generated by default. If the user needs them, the user can specify the `--enable-cds-archive-coh=yes` build configure option.
This change also modified the TestDefaultArchiveLoading.java test.

Testing:

- default build config: classes*coh.jsa CDS archives are not generated
- build configured with `--enable-cds-archive-coh=yes`: classes*coh.jsa CDS archives are generated
- build configured with `--enable-cds-archive-coh=no`: classes*coh.jsa CDS archives are not generated
- passed tiers 1 and 3 tests

Update:

The classes*coh.jsa archives will be generated by default. In case the user doesn't want to generate those archives, just specify the --disable-cds-archive-coh option in the build configure command.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341553](https://bugs.openjdk.org/browse/JDK-8341553): Remove UseCompactObjectHeaders extra CDS archives (**Enhancement** - P3)

### Reviewers
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**) Review applies to [a063f8c2](https://git.openjdk.org/jdk/pull/22051/files/a063f8c289fc9123d3bcab81865812c2d2d01ed6)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) Review applies to [a063f8c2](https://git.openjdk.org/jdk/pull/22051/files/a063f8c289fc9123d3bcab81865812c2d2d01ed6)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22051/head:pull/22051` \
`$ git checkout pull/22051`

Update a local copy of the PR: \
`$ git checkout pull/22051` \
`$ git pull https://git.openjdk.org/jdk.git pull/22051/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22051`

View PR using the GUI difftool: \
`$ git pr show -t 22051`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22051.diff">https://git.openjdk.org/jdk/pull/22051.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22051#issuecomment-2471853012)
</details>

### CODE REVIEW

**Review by iklam — CHANGES_REQUESTED** [review]
I think it's best to avoid modifying the test JVM. It could be used in parallel by another jtreg process, and it could also be in a read-only directory.

I would suggest checking for the existence of the COH files, and skip the test cases accordingly.

**Review by iklam — APPROVED** [review]
Test changes look good.

**Review by coleenp — APPROVED** [review]
Looks good.

File: make/autoconf/jdk-options.m4 [hunk 1]
```diff
@@ -722,9 +722,12 @@ AC_DEFUN([JDKOPT_ENABLE_DISABLE_CDS_ARCHIVE_COH],
              test "x$OPENJDK_TARGET_CPU" != "xs390x"; then
           AC_MSG_RESULT([no (compact object headers not supported for this platform)])
           AVAILABLE=false
-        else
+        elif test "x$enable_cds_archive_coh" = "xyes"; then
           AC_MSG_RESULT([yes])
           AVAILABLE=true
+        else
+          AC_MSG_RESULT([yes])
+          AVAILABLE=false
```
Comment (erikj79) [inline make/autoconf/jdk-options.m4:730]: This does not look correct. The `CHECK_AVAILABLE` block is meant to check if the preconditions for enabling this feature are met. Those preconditions should not change here. All you should need to do is change the default.

**Review by erikj79 — COMMENTED** [review]
I'm a bit unsure what was actually agreed in the bug discussion. Are wanting to disable this by default for OpenJDK, requiring anyone who wants it enabled to add a new configure option, or are we only intending to change the configuration for Oracle builds? If the latter, we should just add the option `--disable-cds-archive-coh` to `jib-profiles.js` for the relevant build profiles.

**Comment by calvinccheung** [comment]
> I'm a bit unsure what was actually agreed in the bug discussion. Are wanting to disable this by default for OpenJDK, requiring anyone who wants it enabled to add a new configure option, or are we only intending to change the configuration for Oracle builds? If the latter, we should just add the option `--disable-cds-archive-coh` to `jib-profiles.js` for the relevant build profiles.

Per our off-line discussion, I went with the latter approach and have changed the `jib-profiles.js`.

File: make/autoconf/jdk-options.m4 (hunk 1 above)
Comment (calvinccheung) [reply make/autoconf/jdk-options.m4]: I've reverted the changes to jdk-optinos.m4.

**Review by erikj79 — APPROVED** [review]

**Comment by calvinccheung** [comment]
Thanks @iklam, @coleenp, @erikj79 for the review.

/integrate

**Pull request closed without merging**